### ISSUE TITLE
Tooltip: support clicking on a tooltip when a reference element is focused.

### ISSUE THREAD

**Issue by driskull** [Esri/calcite-design-system#9827]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

The issue is that when you are focused on a reference element and then click on a tooltip, the onfocusout is fired which closes the open tooltip before the click can occur.

We should have a solution to allow for the tooltip to be clicked and not closed

### Expected Behavior

When focusing a reference element and then clicking a tooltip, the tooltip should not close.

### Reproduction Sample

https://codepen.io/driskull/pen/RwzRKmb?editors=1000

### Reproduction Steps

1. open the console
2. click on the button
3. click on the tooltip
4. the tooltip should remain open

### Reproduction Version

2.10.0

### Relevant Info

Notes: it works if just hovering the reference element and clicking the tooltip.

Basically, the focusout and click are fighting each other in this scenario

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by DitwanP** [comment]
🍡 Verified locally on `dev`
